cli/aws_ddk/services/cfn.py

```diff
@@ -101,13 +106,14 @@ def _execute_changeset(changeset_id: str, stack_name: str) -> None:
 
 
 def _wait_for_execute(stack_name: str, changeset_type: str) -> None:
+    waiter: "Union[StackCreateCompleteWaiter, StackUpdateCompleteWaiter]"
     if changeset_type == "CREATE":
         waiter = boto3_client("cloudformation").get_waiter("stack_create_complete")
     elif changeset_type == "UPDATE":
         waiter = boto3_client("cloudformation").get_waiter("stack_update_complete")
     else:
         raise RuntimeError(f"Invalid changeset type {changeset_type}")
```

Comment:
Ideally, waiters should be separated, because they use different types

```python
    if changeset_type == "CREATE":
        create_waiter = boto3_client("cloudformation").get_waiter("stack_create_complete")
        create_waiter.wait(StackName=stack_name, WaiterConfig=waiter_config)
        return
    if changeset_type == "UPDATE":
        update_waiter = boto3_client("cloudformation").get_waiter("stack_update_complete")
        update_waiter .wait(StackName=stack_name, WaiterConfig=waiter_config)
        return

     raise RuntimeError(f"Invalid changeset type {changeset_type}")
```

Let me know what you think.